### PR TITLE
Add documentation for qutip measurement functions

### PR DESCRIPTION
**Pull request by hodgestar**
This adds:
  - [x] Autodocs for the measurement functions.
  - [x] A guide to measuring states with `measure`.
  - [x] A guide to obtaining measurement statistics with `measurement_statistics`.

### CODE REVIEW

File: guide/guide-measurement.rst [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+.. _measurement-intro:
+
+Introduction
+============
+
+Measurement is a fundamental part of the standard formulation of quantum
+mechanics and is the process by which classical readings are obtained from
+a quantum object. Although the intrepretation of the procedure is at times
+contentious, the procedure itself is mathematically straight forward and is
```
Comment (nathanshammah) [inline guide/guide-measurement.rst:26]:
```suggestion
contentious, the procedure itself is mathematically straightforward and is
```

Comment (nathanshammah) [inline guide/guide-measurement.rst]:
```suggestion
contentious, the procedure itself is mathematically straightforward and is
```

File: guide/guide-measurement.rst [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+.. ipython::
+
+   In [3]: spin_z = sigmaz()
+
+   In [4]: spin_x = sigmax()
+
+How do we know what these operators measure? The answer lies in the measurement
+procedure itself:
+
+* A quantum measurement tranforms the state being measure by projecting it into
```
Comment (nathanshammah) [inline guide/guide-measurement.rst:63]:
```suggestion
* A quantum measurement tranforms the state being measured by projecting it into
```

File: guide/guide-measurement.rst [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+* Which eigenvector to project onto is chosen probabilitically according to the
+  square of the amplitude of the state in the direction of the eigenvector.
+
+* The value returned by the measurement is the eigenvalue corresponding to the
+  chosen eigenvector.
+
+.. note::
+
+   How to intrepret this "random chosing" is the famous
```
Comment (nathanshammah) [inline guide/guide-measurement.rst:74]:
```suggestion
   How to interpret this "random choosing" is the famous
```

File: guide/guide-measurement.rst [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+The `up` state is not an eigenvector of `spin_x`. `spin_x` has two eigenvectors
+which we will call `left` and `right`. The `up` state has equal components in
+the direction of these two vectors, so measurement will select each of them
+50% of the time.
+
+When `left` is chosen, the result of the measurement will be `(1.0, left)`.
+
+When `right` is chosen, the result of measurement with be `(-1.0, right)`.
+
+Now you know how to measure quantum states in QuTip!
```
Comment (nathanshammah) [inline guide/guide-measurement.rst:120]:
```suggestion
Now you know how to measure quantum states in QuTiP!
```

File: guide/guide-measurement.rst [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+The `measure` function can perform measurements on density matrices too. You
+can read about these and other details at :func:`~qutip.measurement.measure`.
+
+.. _measurement-statistics:
+
+Obtaining measurement statistics
+--------------------------------
+
+XXX: TODO
```
Comment (nathanshammah) [inline guide/guide-measurement.rst:130]: Remember to remove the TODO.

**Comment by hodgestar** [comment]
@nathanshammah Thank you for the review and the many small fixes!

I plan to finish the measurement_statistics part of the guide this evening.

File: guide/guide-measurement.rst (hunk 5 above)
Comment (hodgestar) [reply guide/guide-measurement.rst]: Removed!

**Comment by hodgestar** [comment]
@nathanshammah Section on `measurement_statistics` added.

**Comment by hodgestar** [comment]
And now the iPython blocks all run and produce the expected output in my local documentation build.

**Comment by ajgpitch** [comment]
I have merged #qutip/qutip/pull/1090.
@nathanshammah If you are happy with the docs then please go ahead and merge.

**Comment by hodgestar** [comment]
@nathanshammah My process wasn't exactly repeatable -- I hacked the versions of the dependencies I had installed until it worked. I would like to make something repeatable though.

Do you have any thoughts on where to put that? Would Travis builds be the best option to pursue? Or just a Docker container that can do a doc build? Something else? Happy to do the work -- just need a pointer on what direction makes sense for the project.

**Comment by nathanshammah** [comment]
Ok, so basically it’s similar to what I did for release 4.5.0: tweak stuff. I’ll go on with concerting ipython blocks with Python blocks. For the rest, I am not very familiar with building dockers in CI, but I think it’s not even necessary: once the recipe works, we can just add a “make html” and “make doctest” job. I’ve set that up recently on GitHub actions (which I think is emerging very fast as more flexible than Travis and more integrated with the rest of the GH stack) but it will work also with Travis.

**Comment by hodgestar** [comment]
Thanks for landing! :D

**Pull request merged**